### DISCUSSION TITLE
[Feature]: Include Lambda Context in getLoadContext for Architect

### DISCUSSION THREAD

**Discussion by shamsup** [remix-run/remix#3904] [Proposals]
### What is the new or updated feature that you are suggesting?

In @remix-run/architect, the `context` from the second parameter provided to the AWS Lambda handler function should be available in the `getLoadContext` function.

### Why should this feature be included?

Various metadata is included in the secondary `context` parameter provided when a Lambda function is invoked. For example, the `awsRequestId` for tracing, memory configuration, and authentication information if using Cognito as an authorizer in API Gateway,  are inaccessible to the remix application without it being provided to the `getLoadContext` function.

**Comment by zmillman** [comment]
This would be great for me too. Right now I have to do this hacky variable hoisting:

```ts
import { createRequestHandler } from "@remix-run/architect";

// HACK: Variable hoisting to provide request metadata to the Remix handler
// https://github.com/remix-run/remix/discussions/3904
let latestRequestContext;

const handleRequest = createRequestHandler({
  build: build,
  mode: process.env.NODE_ENV,
  getLoadContext: () => {
    return {
      // Properties from Lambda event context
      // https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html
      lambda: latestRequestContext
    };
  },
});

export const handler = async function (event, context) {
    latestRequestContext = context;
    return handleRequest(event, context);
}
```
